client/asset/dash/dash.go

```diff
@@ -0,0 +1,169 @@
+				Description:       "Connect to dashd",
+				DefaultConfigPath: dexbtc.SystemConfigPath("dash"),
+				ConfigOpts:        configOpts,
+			},
+		},
+	}
+)
+
+func init() {
+	asset.Register(BipID, &Driver{})
+}
+
+// Driver implements asset.Driver.
+type Driver struct{}
+
+// Open creates the Dash exchange wallet. Start the wallet with its Run method.
+func (d *Driver) Open(cfg *asset.WalletConfig, logger dex.Logger, network dex.Network) (asset.Wallet, error) {
+	return NewWallet(cfg, logger, network)
+}
+
+// DecodeCoinID creates a human-readable representation of a coin ID for Dash
+func (d *Driver) DecodeCoinID(coinID []byte) (string, error) {
+	// Dash and Bitcoin have the same tx hash and output format.
+	return (&btc.Driver{}).DecodeCoinID(coinID)
+}
+
+// Info returns basic information about the wallet and asset.
+func (d *Driver) Info() *asset.WalletInfo {
+	return WalletInfo
+}
+
+// NewWallet is the exported constructor by which the DEX will import the
+// exchange wallet. The wallet will shut down when the provided context is
+// canceled. The configPath can be an empty string, in which case the standard
+// system location of the dashd config file is assumed.
+func NewWallet(cfg *asset.WalletConfig, logger dex.Logger, network dex.Network) (asset.Wallet, error) {
+	var params *chaincfg.Params
+	switch network {
+	case dex.Mainnet:
+		params = dexdash.MainNetParams
+	case dex.Testnet:
+		params = dexdash.TestNetParams
+	case dex.Regtest:
+		params = dexdash.RegressionNetParams
+	default:
+		return nil, fmt.Errorf("unknown network ID %v", network)
+	}
+
+	// Designate the clone ports. These will be overwritten by any explicit
+	// settings in the configuration file.
+	ports := dexbtc.NetPorts{
+		Mainnet: "9999",  // rpc:  9998
+		Testnet: "19999", // rpc: 19998
+		Simnet:  "19994", // rpc: 19998
+	}
+
+	cloneCFG := &btc.BTCCloneCFG{
+		WalletCFG:           cfg,
+		MinNetworkVersion:   minNetworkVersion,
+		WalletInfo:          WalletInfo,
+		Symbol:              "dash",
+		Logger:              logger,
+		Network:             network,
+		ChainParams:         params,
+		Ports:               ports,
+		DefaultFallbackFee:  dexdash.DefaultFee,
+		DefaultFeeRateLimit: dexdash.DefaultFeeRateLimit,
+		LegacyBalance:       false,
+		ZECStyleBalance:     false,
+		Segwit:              false,
+		// Dash v19.1.0 has a breaking change from the true/false 'allowhighfees'
+		// to 'maxfeerate' in DASH/kB, the same as btc.
+		LegacyRawFeeLimit:        false,
+		InitTxSize:               dexbtc.InitTxSize,
+		InitTxSizeBase:           dexbtc.InitTxSizeBase,
+		PrivKeyFunc:              nil,
+		AddrFunc:                 nil,
+		AddressDecoder:           nil,
+		AddressStringer:          nil,
+		BlockDeserializer:        nil,
+		ArglessChangeAddrRPC:     true, // getrawchangeaddress has No address-type arg
+		NonSegwitSigner:          nil,
+		ConnectFunc:              nil,
+		FeeEstimator:             nil,   // estimatesmartfee + getblockstats
```

Comment:
You mean `ExternalFeeEstimator` correct ?

I will implement this similarly to Firo, Doge. I will also look for billing information ..worst case is we have an implementation and we do not use it.

Thanks for the link

Reply:
[API](https://blockchair.com/api/docs)
"_Our API is free to try under some limitations_, and we have a variety of premium plans. Please check out the information about [the limits and plans](https://blockchair.com/api/docs#link_M05)."
So bottom line looks like we can get __"Up to 1440 requests a day"__ with no API key

Reply:
So that's one request per minute. I guess we need to cache our external fee rates.

Reply:
Implemented `dash.fetchExternalFee`

Reply:
I will think about caching but pushing the naive impl for this PR

Reply:
Allow 1 web access per minute, else cache latest good value:

Maybe something using [time.NewTicker](http://golang.org/pkg/time/#NewTicker) and (global?) state .. any suggestions where to keep the latest value off the top of your head ;-)
In the client wallet?

Reply:
Okay I have slept on this. Originally I had thought that the ideal solution was a central server proxying the calls to blockchair. But even I always thought that was way over the top ;-)

So I guess each dash client can cache their own. They each have an unique IP address to which the 1440 requests would be attributed.

So I will fix that today and then request another 'review'

Reply:
Updated with
[Get external fee from blockchair with local cache](https://github.com/decred/dcrdex/pull/2380/commits/361d09864b9a8df7e20c940eb3309ef66ffc37b6)

This caches locally. It could also be refactored later to get stats for the other manstrem coins blockchair supports: e.g. btc, zec, ltc, always with the proviso of 1440 hits per day per IP address.
[supported coins](https://blockchair.com/api/docs#link_M1)